pkg/client/fetch_summary/fetch_summary.go

```diff
@@ -72,6 +85,28 @@ func (c *Client) SaveFetchSummary(ctx context.Context, fs *FetchSummary) error {
 	if err != nil {
 		return err
 	}
-
 	return c.db.Exec(ctx, sql, args...)
 }
+
+// GetLatestFetchSummaryForProvider gets latest fetch summary for specific provider
+func (c *FetchSummaryClient) GetLatestFetchSummaryForProvider(ctx context.Context, provider string) (*FetchSummary, error) {
+	q := goqu.Dialect("postgres").
+		Select("provider_version", "is_success").
+		From("cloudquery.fetches").
+		Where(goqu.Ex{"provider_name": provider, "finish": goqu.Op{"isNot": nil}}).
+		Limit(1).
+		Order(goqu.I("finish").Desc())
+	sql, _, err := q.ToSQL()
+	if err != nil {
+		return nil, err
+	}
+	var data []FetchSummary
+	err = pgxscan.Select(ctx, c.db, &data, sql)
```

Comment:
If selecting a single row, could do `.Get` instead. No rows would return an error (pgx.ErrNoRows but wrapped)

But maybe get latest row per version? Not sure